core/src/toga/app.py

```diff
@@ -91,6 +91,27 @@ def discard(self, window: Window) -> None:
             raise ValueError(f"{window!r} is not part of this app")
         self.elements.remove(window)
 
+    ######################################################################
+    # 2023-10: Backwards compatibility
+    ######################################################################
+
+    def __iadd__(self, window: Window) -> None:
+        # The standard set type does not have a += operator.
+        warn("Instead of +=, use add()", DeprecationWarning, stacklevel=2)
+        self.add(window)
```

Comment:
Since any user-space usage of += or -= is going to be fighting against Toga's internal window management, my inclination is to make these two no-ops - retain the API, but raise a warning that leads the user to you don't need to do this at all, rather than suggesting they need to use a different API name.